runners/spark/src/main/java/org/apache/beam/runners/spark/translation/SparkRuntimeContext.java

```diff
@@ -118,6 +117,24 @@ public CoderRegistry getCoderRegistry() {
     return coderRegistry;
   }
 
+  private static class PipelineOptionsHolder {
+    // on executors, this should deserialize once.
+    private static transient volatile PipelineOptions pipelineOptions = null;
+
+    static PipelineOptions getOrInit(String serializedPipelineOptions) {
+      if (pipelineOptions == null) {
+        synchronized (PipelineOptionsHolder.class) {
+          if (pipelineOptions == null) {
+            pipelineOptions = deserializePipelineOptions(serializedPipelineOptions);
+          }
+        }
+        // register IO factories.
+        IOChannelUtils.registerIOFactoriesAllowOverride(pipelineOptions);
```

Comment:
Good question!
From Spark docs: "Each application gets its own executor processes".
[This](http://spark.apache.org/docs/1.6.3/cluster-overview.html) page in Spark docs should cover it a lot better than me.
So I think a lazy-singleton should work (pretty common pattern for Spark users..).